internal/test_cases/file_search_test_case.go

```diff
@@ -0,0 +1,81 @@
+package test_cases
+
+import (
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+type FileSearchTestCase struct {
+	Pattern          string
+	FilePaths        []string
+	ExpectedExitCode int
+	ExpectedOutput   []string
+	Recursive        bool
+}
+
+func (testCase FileSearchTestCase) Run(stageHarness *test_case_harness.TestCaseHarness) error {
+	return runFileSearchTestCases([]FileSearchTestCase{testCase}, stageHarness)
```

Comment:
Pointless indirection, why aren't we just placing the code here instead of adding a private function to house it?